### PR TITLE
Add new functions to Math MK2 node: mean and gcd (greatest common divisor)

### PR DESCRIPTION
**Pull request by DolphinDream**
## Addressed problem description

Add some missing functions: **arithmetic mean**: (x + y) /2 and **gdc** (greatest common divisor).

## Preflight checklist

- [x] Code changes complete.
- [x] Code documentation complete.
- [x] Documentation for users complete (or not required, if user never sees these changes).
- [x] Manual testing done. 
- [ ] Unit-tests implemented.
- [x] Ready for merge.

### CODE REVIEW

**Comment by DolphinDream** [comment]
I’m not sure about the function indexing. I did my best to put the new function in a reasonable place in th elist, but if a different location is preferred, please advise.

File: nodes/number/scalar_mk2.py [hunk 1]
```diff
@@ -72,6 +73,8 @@
     "ROUND-N":     (81,  lambda x, y: round(x, int(y)),   ('ss s'), "Round N",),
     "FMOD":        (82,  fmod,                            ('ss s'), "Fmod"),
     "MODULO":      (83,  lambda x, y: (x % y),            ('ss s'), "modulo"),
+    "MEAN":        (84,  lambda x, y: 0.5*(x + y),        ('ss s'), "mean"),
+    "GCD":         (85,  lambda x, y: gcd(x,y),           ('ss s'), "gcd"),
```
Comment (zeffii) [inline nodes/number/scalar_mk2.py]: 👍

Comment (zeffii) [reply nodes/number/scalar_mk2.py]: but i think you can just do 
```
"GCD":         (85,  gcd,           ('ss s'), "gcd"),
```

**Comment by zeffii** [comment]
done

**Pull request merged**